src/lib/y2storage/devicegraph.rb

```diff
@@ -559,5 +557,79 @@ def udev_lookup_possible?
       # called provides a seasonable result.
       !StorageManager.instance.committed?
     end
+
+    # Encryption devices might be probed with a name that does not match the device name
+    # indicated in the fstab file. For example, /etc/fstab could have an entry like:
+    #
+    #   /dev/mapper/cr_home   /home   ext4  defaults  0   0
+    #
+    # But that encryption device could be probed with a name like /dev/mapper/cr-auto-1. In that
+    # case, the device could not be found in the devicegraph when searching for the device name in the
+    # fstab entry. But, if the crypttab file was previously parsed (see Encryption#save_crypttab_names),
+    # the Encryption devices are populated in the devicegraph with their corresponding name indicated
+    # in the crypttab. This information can be used to try possible alternative names for the encryption
+    # device. For example, when the devicegraph contains a Encryption layer /dev/mapper/cr-auto-1 over
+    # the device /dev/sda1, and the /etc/crypttab has the following entry:
+    #
+    #   cr_home   /dev/sda1
+    #
+    # a possible alternative name for /dev/mapper/cr_home would be /dev/mapper/cr-auto-1, due to the
+    # encryption device cr-auto-1 has "cr_home" as crypttab_name (after parsing the crypttab file).
+    #
+    # @param device_name [String] a kernel name or udev name
+    # @return [Array<String>]
+    def alternative_names(device_name)
+      devices = encryptions.select { |e| e.crypttab_name? && device_name.match?(e.crypttab_name) }
+
+      devices.map { |d| device_name.sub(d.crypttab_name, d.dm_table_name) }
+    end
+
+    # Performs the search of a device by any possible name (kernel name or udev name)
+    #
+    # This is different from {BlkDevice.find_by_any_name} in several ways:
+    #
+    # * It will find any matching device, not only block devices (e.g. LVM VGs
+    #   also have a name but are not block devices).
+    # * It can be called on any devicegraph, not only probed.
+    # * It uses a system lookup only when necessary (i.e. all the cheaper
+    #   methods for finding the device have been unsuccessful).
+    # * It avoids system lookup in potentially risky scenarios (like an outdated
+    #   {StorageManager#probed}).
```

Comment:
A lot of documentation repeated from the public method. A simple `@see #find_by_any_name` would be enough for this internal method. :-)